fsmonitor.c

```diff
@@ -32,17 +33,27 @@ int read_fsmonitor_extension(struct index_state *istate, const void *data,
 	uint32_t ewah_size;
 	struct ewah_bitmap *fsmonitor_dirty;
 	int ret;
+	uint64_t tm;
+	struct strbuf last_update = STRBUF_INIT;
 
 	if (sz < sizeof(uint32_t) + sizeof(uint64_t) + sizeof(uint32_t))
 		return error("corrupt fsmonitor extension (too short)");
 
 	hdr_version = get_be32(index);
 	index += sizeof(uint32_t);
-	if (hdr_version != INDEX_EXTENSION_VERSION)
+	if (hdr_version == INDEX_EXTENSION_VERSION1) {
+		tm = get_be64(index);
+		strbuf_addf(&last_update, "%"PRIu64"", tm);
+		index += sizeof(uint64_t);
+	} else if (hdr_version == INDEX_EXTENSION_VERSION2) {
+		strbuf_addstr(&last_update, index);
+		index += last_update.len + 1;
+	} else {
+		strbuf_release(&last_update);
```

Comment:
No need to release that here, we have not allocated anything yet (`STRBUF_INIT` is a static initializer, it cannot allocate any memory).

Reply:
For some reason I was thinking that every `STRBUF_INIT` needed a matching `strbuf_release` so there was a memory leak but if not all the better thanks!